logstash-core/lib/logstash/util/wrapped_synchronous_queue.rb

```diff
@@ -73,7 +80,7 @@ def close
       end
 
       def empty?
-        true # synchronous queue is alway empty
+        @queue.queue.isEmpty
```

Comment:
Does this work? If `@queue` is an ArrayBlockingQueue. there is no such `queue` method that I can see (https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ArrayBlockingQueue.html). If it is a SynchronousQueue, I cannot find any such `queue` method either (https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/SynchronousQueue.html)